pennylane/tensorbox/tensorbox.py

```diff
@@ -0,0 +1,253 @@
+
+        namespace = tensor.__class__.__module__.split(".")[0]
+
+        if isinstance(tensor, (list, tuple)) or namespace == "numpy":
+            from .numpy_box import NumpyBox
+
+            return NumpyBox.__new__(NumpyBox, tensor)
+
+        if namespace in ("pennylane", "autograd"):
+            from .autograd_box import AutogradBox
+
+            return AutogradBox.__new__(AutogradBox, tensor)
+
+        if namespace == "tensorflow":
+            from .tf_box import TensorFlowBox
+
+            return TensorFlowBox.__new__(TensorFlowBox, tensor)
+
+        if namespace == "torch":
+            from .torch_box import TorchBox
+
+            return TorchBox.__new__(TorchBox, tensor)
+
+        raise ValueError(f"Unknown tensor type {type(tensor)}")
+
+    def __init__(self, tensor):
+        self._data = tensor
+
+    def __repr__(self):
+        return f"<TensorBox {self.unbox().__repr__()}>"
+
+    def __len__(self):
+        return len(self.unbox())
+
+    def __mul__(self, other):
+        if isinstance(other, TensorBox):
+            other = other.unbox()
+
+        return self.__class__(self.unbox() * other)
+
+    def __pow__(self, other):
+        if isinstance(other, TensorBox):
+            other = other.unbox()
+
+        return self.__class__(self.unbox() ** other)
+
+    def __rpow__(self, other):
+        return self.__class__(other ** self.unbox())
+
+    __rmul__ = __mul__
+
+    @staticmethod
+    def unbox_list(tensors):
+        """Unboxes or unwraps a list of array like objects, converting any :class:`TensorBox`
+        objects in the list into raw interface tensors.
+
+        Args:
+            tensors (list[array_like]): list of arrays, tensors, or :class:`~.TensorBox` objects
+
+        Returns
+            list[array_like]: the input list with all :class:`TensorBox` objects
+            unwrapped.
+
+        **Example**
+
+        >>> x = tf.Variable([0.4, 0.1, 0.5])
+        >>> y = TensorBox(x)
+        >>> z = tf.constant([0.1, 0.2])
+
+        Note that this is a static method, so we must pass the unified tensors itself
+        if we would like it to be included.
+
+        >>> res = y.unwrap([y, z])
+        >>> res
+        [<tf.Variable 'Variable:0' shape=(3,) dtype=float32, numpy=array([0.4, 0.1, 0.5], dtype=float32)>,
+         <tf.Tensor: shape=(2,), dtype=float32, numpy=array([0.1, 0.2], dtype=float32)>]
+        >>> print([type(v) for v in res])
+        [<class 'tensorflow.python.ops.resource_variable_ops.ResourceVariable'>,
+         <class 'tensorflow.python.framework.ops.EagerTensor'>]
+        """
+        return [v.unbox() if isinstance(v, TensorBox) else v for v in tensors]
+
+    def unbox(self):
+        """Unboxes the ``TensorBox`` container, returning the raw interface tensor."""
+        return self._data
+
+    @property
+    @abc.abstractmethod
+    def interface(self):
+        """str, None: The linear algebra package that the :class:`.TensorBox` class

```

Comment:
Maybe better just to say "package" since it is not all linear algebra they do?

Reply:
```suggestion
        """str, None: The package that the :class:`.TensorBox` class
```